tests/test_questionnaire_validator.py

```diff
@@ -406,3 +406,74 @@ def test_invalid_calculated_or_grand_calculated_summary_id_in_value_source():
     validator.validate()
 
     assert validator.errors == expected_errors
+
+
+def test_list_as_source_referenced_before_created():
+    filename = "schemas/invalid/test_invalid_list_source_reference.json"
+    validator = QuestionnaireValidator(_open_and_load_schema_file(filename))
+
+    expected_errors = [
+        {
+            "message": "List 'household' referenced as source before it has been "
```

Comment:
V minor/cosmetic - each `message` on a single line?

Reply:
They are now are after we removed list name to it's own key/value pair, thanks.

Reply:
We can pass the actual error message constant here now, import it, but only when they list id key/val is separated out so not sure if we want to separate it out? @berroar?

Reply:
I  think splitting it out is fine (e.g. the current one) because at the end of the day, it is a case of clearly presenting the info to an author, so looks good to me 👍

Reply:
Ok, added the changes now, thanks.